### PR TITLE
feat(Computable): generalize ComputableInPolyTime

### PR DESCRIPTION
**Pull request by hmonroe**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Computability/TMComputable.lean [hunk 1]
```diff
@@ -315,6 +315,21 @@ instance inhabitedTM2ComputableAux : Inhabited (TM2ComputableAux Bool Bool) :=
...
+/-- Indicator function for a language decidable in polynomial time -/
+def PolyTimeLanguageIndicator {α: Type} {ea : FinEncoding α} {ebool : FinEncoding Bool}
+    (f : α → Bool) := @TM2ComputableInPolyTime α Bool ea ebool f
+
+/-- Relationship indicates that "x" has a certificate with encoded length polynomially bounded as
+a function of the encoded length of x. -/
+def verification {α γ: Type} {ea : FinEncoding α} {eg : FinEncoding γ} {p: Polynomial ℕ}
+    (certified : Rel α γ) : Prop :=
+  ∀ (x : α) (certificate : γ), certified x certificate →
+  (ea.encode x).length > p.eval (eg.encode certificate).length
```
Comment (alreadydone) [inline Mathlib/Computability/TMComputable.lean:331]: Why is this not `eg.encode certificate < p.eval (ea.encode x).length`

Comment (hmonroe) [reply Mathlib/Computability/TMComputable.lean]: Please see the correction--it is the certificate length that must be bounded as a function of the length of x. Thanks for pointing that out.

**Comment by madvorak** [comment]
Can you please add some unit tests?

**Comment by hmonroe** [comment]
> Can you please add some unit tests?

Could you please clarify what you would like to be tested and how?

**Comment by madvorak** [comment]
> > Can you please add some unit tests?
> 
> Could you please clarify what you would like to be tested and how?

Any concrete example where `verification` holds.

Adding definitions without theorems about them is IMO ok, but you should provide an example that shows how the definitions work.

**Pull request closed without merging**